### ISSUE TITLE
wav2vec2 padding issue

### ISSUE THREAD

**Issue by jeffxtang** [pytorch/fairseq#3278]
## 🐛 Bug

I'm trying to use the https://github.com/pytorch/fairseq/tree/master/examples/wav2vec model on mobile, as I can trace but not script the model, I tried using padding to allow for different audio inputs. But calling `Wav2Vec2Tokenizer` with `max_length` padding results in bad recognition results if `max_length` gets larger.

### To Reproduce

Steps to reproduce the behavior (**always include the command you ran**):

Below is the code and results - padding a 1.4s wav to 3s or 5s lead to about the same recognition, but to 7s or 9s lead to worse and worse results:

```
import soundfile as sf
import torch
from transformers import Wav2Vec2ForMaskedLM, Wav2Vec2Tokenizer

tokenizer = Wav2Vec2Tokenizer.from_pretrained("facebook/wav2vec2-base-960h")
model = Wav2Vec2ForMaskedLM.from_pretrained("facebook/wav2vec2-base-960h")
model.eval()
audio_input, _ = sf.read("ionlywishtobealone.wav")
input_values = tokenizer(audio_input, return_tensors="pt").input_values
logits = model(input_values).logits
predicted_ids = torch.argmax(logits, dim=-1)
transcription = tokenizer.batch_decode(predicted_ids)[0]
print(transcription)
print(input_values.shape)

# pad the 1.42s audio to about 3,5,7,9s:
for n in [3,5,7,9]:
  input_values = tokenizer(audio_input, return_tensors="pt", padding="max_length", max_length=n*10000).input_values
  logits = model(input_values).logits
  predicted_ids = torch.argmax(logits, dim=-1)
  transcription = tokenizer.batch_decode(predicted_ids)[0]
  print(input_values.shape)
  print("padding to {}s: {}\n".format(n, transcription))
```

```
I ONLY WAS TO BE ALONE
torch.Size([1, 15616])

torch.Size([1, 30000])
padding to 3s: I ONLY RIS TO BE ALONE

torch.Size([1, 50000])
padding to 5s: I ONLY IS TO BE ALONE

torch.Size([1, 70000])
padding to 7s: ONER IS TO BE LON

torch.Size([1, 90000])
padding to 9s: ONERISTO BELAN
```

### Expected behavior
Padding to longer audio should lead to the same results.

### Environment

 - fairseq Version (e.g., 1.0 or master): NA
 - PyTorch Version (e.g., 1.0) 1.7 & master
 - OS (e.g., Linux): Mac
 - How you installed fairseq (`pip`, source): NA
 - Build command you used (if compiling from source): NA
 - Python version: 3.8
 - CUDA/cuDNN version: NA
 - GPU models and configuration: NA
 - Any other relevant information:

**Comment by jeffxtang** [comment]
[ionlywishtobealone.wav.zip](https://github.com/pytorch/fairseq/files/6039624/ionlywishtobealone.wav.zip)

**Comment by jeffxtang** [comment]
Fixed in the update of the script [here](https://github.com/pytorch/android-demo-app/tree/master/SpeechRecognition) using torchaudio.
